coverage.sh

```diff
@@ -1,12 +1,5 @@
 #!/usr/bin/env bash
 
 set -e
-echo "" > coverage.txt
-
-for d in $(go list ./... | grep -v vendor); do
-    go test -race -coverprofile=profile.out -covermode=atomic $d
-    if [ -f profile.out ]; then
-        cat profile.out >> coverage.txt
-        rm profile.out
-    fi
-done
+go list ./... | grep -v vendor | parallel -k go test -race -coverprofile=out/{#} -covermode=atomic {}
```

Comment:
On further research, parallel is actually better because it schedules jobs on cpus efficiently that fully utilizes the available resources without starving the host machine of cpu. 

It's also possible to preserve the `nosudo` flag by utilizing travis's apt addon so we can keep using the faster containerized build infrastructure.